app/validation/validator.py

```diff
@@ -200,8 +200,16 @@ def validate_when_rule(when_clause, existing_ids, referenced_id):
         answer_reference_id_fields = ('id', 'answer_count')
 
         for when in when_clause:
-            # either of the reference fields may be in use
-            id_ref_key = [x for x in answer_reference_id_fields if x in when].pop()
+            # either of the reference fields may be in use, however it will
+            # never be the case that both are defined since this is handled by
+            # the JSON schema check
+            present_id_ref_keys = [x for x in answer_reference_id_fields if x in when]
+
+            if not len(present_id_ref_keys):
```

Comment:
Another minor one - unnecessary call to `len`. Empty lists evaluate to `False` anyway.

Not blocking.

Reply:
Made the change in any case